### PR TITLE
feat(metadata): expose client's algolia agent

### PR DESCRIPTION
**Pull request by Haroenv**
**Summary**

The user agent can be used to inspect whether the client is using InstantSearch.js via an integration like Vue InstantSearch or Angular InstantSearch

**Result**

metadata now exposes the algolia agent

### CODE REVIEW

File: tsconfig.v3.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "extends": "./tsconfig",
+  "exclude": [
+    "examples",
+    "es",
+    // this test has specific code for v3 and v4, so already checked in the v4 test
```
Comment (eunjae-lee) [inline tsconfig.v3.json:6]: This comment doesn't break anything?

File: src/middlewares/__tests__/createMetadataMiddleware.ts [hunk 2]
```diff
@@ -110,45 +112,220 @@ describe('createMetadataMiddleware', () => {
...
+        });
+
+        search.start();
+
+        await wait(100);
+
+        expect(document.head).toMatchInlineSnapshot(`
+          <head>
+            <meta
+              content="{\\"widgets\\":[],\\"ua\\":\\"Algolia for JavaScript (4.8.6); Node.js (12.14.1); instantsearch.js (4.17.0); JS Helper (3.4.4)\\"}"
```
Comment (eunjae-lee) [inline src/middlewares/__tests__/createMetadataMiddleware.ts:177]: Can we check these with regex with only actual number on the major?
Every time we update the version of any of them, this test will fail.

File: tsconfig.v3.json (hunk 1 above)
Comment (Haroenv) [reply tsconfig.v3.json:6]: no, tsconfig allows comments

File: src/middlewares/__tests__/createMetadataMiddleware.ts (hunk 2 above)
Comment (Haroenv) [reply src/middlewares/__tests__/createMetadataMiddleware.ts]: solved in afc78a6

**Review by eunjae-lee — APPROVED** [review]
Looks good to me.

**Pull request merged**